### PR TITLE
Support aseprite

### PR DESCRIPTION
**Pull request by rsuu**
[aseprite header](https://github.com/aseprite/aseprite/blob/main/docs/ase-file-specs.md#header)

```text
All sorted in little-endian

0..4    u32    File size
4..6    u16    Magic number (0xA5E0)
6..8    u16    Frames 
8..10   u16    Width in pixels
10..12  u16    Height in pixels
```

### CODE REVIEW

**Comment by Roughsketch** [comment]
I really appreciate the PR, but I'm not sure if this belongs in a general use library. I've never heard of this format before and it seems to be a custom format for a paid program and not an open standard.

Is there some history of common usage for this format outside the program?

**Comment by rsuu** [comment]
> I really appreciate the PR, but I'm not sure if this belongs in a general use library. I've never heard of this format before and it seems to be a custom format for a paid program and not an open standard.
> 
> Is there some history of common usage for this format outside the program?

1. Aseprite is open source.
Build from source: https://github.com/aseprite/aseprite/blob/main/INSTALL.md#compiling

2. > The FLIC animation format was originally developed by Autodesk for use with Autodesk Animator (FLI) and Autodesk Animator Pro (FLC). In 1993 Jim Kent had a Dr. Dobbs article with a source code listing, introducing the FLIC format to the public. [^1]

More information: https://github.com/AnimatorPro/Animator-Pro#animator-pro-aka

In GIMP: https://gitlab.gnome.org/GNOME/gimp/-/tree/master/plug-ins/file-fli

---
But .ase and .flc are a bit different.
> The format is much like FLI/FLC files, but with different magic number and differents chunks. [^2]

[^2]: https://github.com/aseprite/aseprite/blob/main/docs/ase-file-specs.md#introduction

[^1]: https://en.m.wikipedia.org/wiki/FLIC_(file_format)

File: src/lib.rs [hunk 1]
```diff
@@ -108,7 +109,11 @@ pub fn image_type(header: &[u8]) -> ImageResult<ImageType> {
         || header.starts_with(b"\x00\x00\x00\x0CJXL \x0D\x0A\x87\x0A")
     {
         Ok(ImageType::Jxl)
+    } else if header.len() >= 12 && &header[4..6] == b"\xE0\xA5" {
+        Ok(ImageType::Aseprite)
     } else {
+        println!("{:?}", &header);
```
Comment (Roughsketch) [inline src/lib.rs:115]: Remove extra print

File: src/lib.rs [hunk 2]
```diff
@@ -260,7 +266,7 @@ fn bmp_size<R: BufRead + Seek>(reader: &mut R) -> ImageResult<ImageSize> {
 
 fn gif_size(header: &[u8]) -> ImageResult<ImageSize> {
     Ok(ImageSize {
-        width:  ((header[6] as usize) | ((header[7] as usize) << 8)),
+        width: ((header[6] as usize) | ((header[7] as usize) << 8)),
```
Comment (Roughsketch) [inline src/lib.rs:269]: To minimize the diff I'd appreciate removing smaller format only changes like this. I can run a cargo fmt at some point in the future.

Not going to tag every single instance, but there are lots of diffs with just spacing changes. For reference I count 11 blocks of them including this one.

**Review by Roughsketch — CHANGES_REQUESTED** [review]
Just a heads up that although this PR is pretty much fine, I'd appreciate if you could handle some small review changes.

The bigger thing is since #22 was merged your tests will need to be changed slightly. Basically just move `test/aseprite` into `tests/images/aseprite` and modify your test paths accordingly.

If for some reason you cannot do this before the next release I have no issue cleaning it up myself so don't feel pressured to do it ASAP. And as always thanks for the PR!

**Comment by rsuu** [comment]
> Just a heads up that although this PR is pretty much fine, I'd appreciate if you could handle some small review changes.
> 
> The bigger thing is since #22 was merged your tests will need to be changed slightly. Basically just move `test/aseprite` into `tests/images/aseprite` and modify your test paths accordingly.
> 
> If for some reason you cannot do this before the next release I have no issue cleaning it up myself so don't feel pressured to do it ASAP. And as always thanks for the PR!

ok, files has been renamed and the code has been removed now.

**Review by Roughsketch — APPROVED** [review]

**Pull request merged**